fw/http_frame.c

```diff
@@ -2208,18 +2211,32 @@ tfw_h2_make_frames(struct sock *sk, TfwH2Ctx *ctx, unsigned long snd_wnd,
 	bool error_was_sent = false;
 	int r = 0;
 
-	while (tfw_h2_stream_sched_is_active(&sched->root)
+#define SCHED_REMOVE_EXCLUSIVE_STREAM(sched, stream)			\
+do {									\
+	if (!tfw_h2_stream_is_exclusive(stream)) {			\
+		parent = stream->sched.parent;				\
+		tfw_h2_stream_sched_remove(sched, stream);		\
+	} else {							\
+		parent = NULL;						\
+	}								\
```

Comment:
The macro is named `SCHED_REMOVE_EXCLUSIVE_STREAM`, but `tfw_h2_stream_sched_remove()` is called for `!tfw_h2_stream_is_exclusive(stream)`